p2p/src/net/libp2p/mod.rs

```diff
@@ -70,6 +71,9 @@ pub struct Libp2pSocket {
     /// Multiaddress of the remote peer
     addr: Multiaddr,
 
+    /// Unique ID of the peer
+    pub id: PeerId,
```

Comment:
pub?

Reply:
I'll check if `pub` can be removed but basically our own peer ID is stored inside the the `Libp2pService` and having access to it is IMO cleaner than returning `(Libp2pService, PeerId)`. I also don't like the idea of "polluting" the `NetworkService` trait with getters.

Reply:
It's actually not pollution, I'd argue. Though I sympathize with what you're thinking. There are crates that add getters by just adding an attribute if you're worried about the lines of code. 

Using a getter basically gives you compile-time guarantees, including, for example, the inability to mutate the value even if the struct was mutable.

Reply:
Done, `pub` was not needed